cpp/src/hdbscan/detail/membership.cuh

```diff
@@ -76,13 +76,18 @@ void get_probabilities(const raft::handle_t& handle,
   rmm::device_uvector<value_t> deaths(n_clusters, stream);
   thrust::fill(exec_policy, deaths.begin(), deaths.end(), 0.0f);
 
+  cudaError_t (*reduce_func)(void*,
```

Comment:
Why was the change to a function pointer needed?

Reply:
I'm not sure. CUB was failing to compile the previous code because of some API change. @hcho3 Do you have anything to add here?

@divyegala We may have to merge this as-is and revisit the question, or else CI will be blocked once the rest of RAPIDS is updated to CCCL 2.2.0.

Reply:
The change was made in [`b1302bd` (#5623)](https://github.com/rapidsai/cuml/pull/5623/commits/b1302bd0f808a881c85bbe90f8c3fabe2a0e1fb7) but the commit message only says "Fix build."

Reply:
Okay, it's not egregious so I will approve for now.

Reply:
I kept getting the build error like
```
error: no instance of function template "detail::Utils::cub_segmented_reduce" matches the argument list
            argument types are: (float *, float *, int, int *, cudaStream_t, <unknown-type>)
```
because the compiler failed to build the type of `cub::DeviceSegmentedReduce::Max`. I had to explicitly declare the type of `cub::DeviceSegmentedReduce::Max` by declaring a function pointer.